spec/classes/app_stack_spec.rb

```diff
@@ -42,17 +77,23 @@
         it { is_expected.to  contain_file('/tmp/ui-cert.key').that_comes_before('Docker_compose[hdp]') }
         it { is_expected.to  contain_file('/tmp/ui-cert.pem').that_comes_before('Docker_compose[hdp]') }
         it {
-          is_expected.to contain_docker_compose('hdp').that_subscribes_to(
-            [
-              'File[/tmp/ui-ca.pem]',
-              'File[/tmp/ui-cert.key]',
-              'File[/tmp/ui-cert.pem]',
-              # 'File[/opt/puppetlabs/hdp/docker-compose.yaml]',
-            ],
-          )
+          is_expected.to contain_docker_compose('hdp')
+            .with_compose_files(['/opt/puppetlabs/hdp/docker-compose.yaml'])
+            .that_requires('File[/opt/puppetlabs/hdp/docker-compose.yaml]')
+            .that_subscribes_to(
+              [
+                'File[/tmp/ui-ca.pem]',
+                'File[/tmp/ui-cert.key]',
+                'File[/tmp/ui-cert.pem]',
+                'File[/opt/puppetlabs/hdp/docker-compose.yaml]',
```

Comment:
Asking for mostly learning puppet purposes - this docker_compose resource requires only the docker-compose file, but subscribes to the cert and key files as well. Should it also require these files if its subscribed to them? Or because it doesn't require them under certain config options, we don't include that here?

Reply:
That is a very good and valid question. That was being tested with these lines from above:

```ruby
        it { is_expected.to  contain_file('/tmp/ui-cert.key').that_comes_before('Docker_compose[hdp]') }
        it { is_expected.to  contain_file('/tmp/ui-cert.pem').that_comes_before('Docker_compose[hdp]') }
```

I have made this more clear by combining those lines into this block and will push it in just a moment

Reply:
This change has been pushed here